lib/cn/csched_sp3.c

```diff
@@ -580,7 +581,8 @@ sp3_refresh_thresholds(struct sp3 *sp)
         thresh.rspill_wlen_max = ((v >> 16) & 0xffff) << 20;
     } else {
         thresh.rspill_runlen_max = SP3_RSPILL_RUNLEN_MAX_DEFAULT;
-        thresh.rspill_runlen_min = SP3_RSPILL_RUNLEN_MIN_DEFAULT;
+        thresh.rspill_runlen_min = sp->rp->csched_full_compact ? 1 :
+                                   SP3_RSPILL_RUNLEN_MIN_DEFAULT;
```

Comment:
In some cases we use adjust other compaction params as we do here.  In other cases we use `csched_full_compact` directly (I will point out an example in another comment).  I think it would be better to rely one approach:  modify existing params or use new param, but not a mix.

Advantage of relying on existing params:
- You can use full compaction to observe natural compaction logic, which could point out inefficiencies in compaction logic.  This is a tiny advantage bc the natural logic changes when the parameters change.
- Less invasive to existing compaction logic

Advantages of relying on a single "full compact" paramter:
- Trivial to turn off if we ever want to support online full compaction.
- Can take a more direct path toward full compaction.
- Less like to be broken by future code changes.

Reply:
modify existing params has my vote